src/main/routes/home.ts

```diff
@@ -1,7 +1,93 @@
+import * as fs from 'fs';
+import * as path from 'path';
+
+import { SecureRequester } from './../assets/js/secureRequester';
+
+import axios from 'axios';
 import { Application } from 'express';
 
+const multer = require('multer');
+
 export default function (app: Application): void {
-  app.get('/', (req, res) => {
-    res.render('home');
+  const upload = multer({ dest: 'uploads/' });
+
+  app.get('/', async (req, res) => {
+    try {
+      const data = { 'output': 'Ready...' };
+      res.render('home', { data });
+    } catch (error) {
+      console.error('Error making request:', error);
+      res.render('home', {});
+    }
+  });
+
+  app.post('/', upload.single('myfile'), async (req, res) => {
+
+    if (!req.file) {
+      return res.status(400).send('No file uploaded');
+    }
+
+    //may need globalprotect on and f5 off else it can't get sas token
+    const output = 'Starting...';
+    const data = { output };
+    try {
+      const { environment, jurisdiction } = req.body;
+
+      // ----------------------START: Grab the sas token------------------------
+      let sasToken = '';
+      const requester = new SecureRequester(environment);
+      const response = await requester.getRequest('/reform-scan/token/' + jurisdiction);
+      if (response) {
+        console.log('SAS token:', response.data.sas_token);
+        data.output += '\nSAS token: ' + response.data.sas_token;
+        sasToken = response.data.sas_token;
+      }
+      // ----------------------END: Grab the sas token------------------------
+      // ----------------------START: Generate a random file name------------------------
+
+      const randomNumbers = Array.from({ length: 13 }, () => Math.floor(Math.random() * 10)).join('');
```

Comment:
It needs to be within the same block ideally, more so for coding principles. Not the end of the world but please change if possible as it would only take a minute to address

Reply:
changed